src/taskManager.js

```diff
@@ -0,0 +1,138 @@
+        check.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2m-6 9l2 2 4-4" />
+        </svg>`;
+      }
+
+      check.addEventListener('click', () => {
+        li.classList.toggle('completed');
+        if (li.classList.contains('completed')) {
+          check.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2m-6 9l2 2 4-4" />
+          </svg>`;
+          task.completed = taskStatus.updateStatus(li);
+          this.setStore();
+        } else {
+          check.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2" />
+          </svg>`;
+          task.completed = taskStatus.updateStatus(li);
+          this.setStore();
+        }
+      });
+
+      dots.addEventListener('click', (e) => {
+        e.stopPropagation();
+        li.classList.add('edit');
+        inputField.focus();
+
+        if (li.classList.contains('edit')) {
+          li.addEventListener('keydown', (e) => {
+            if (e.key === 'Enter' && inputField.value.trim()) {
+              task.description = inputField.value.trim();
+              this.setStore();
+              this.displayTasks(tasksList);
+              inputField.blur();
+              li.classList.remove('edit');
+            }
+          });
+        }
+      });
+
+      inputField.addEventListener('mouseout', (e) => {
+        e.stopPropagation();
+        if (inputField.value.trim()) {
+          task.description = inputField.value.trim();
+          this.setStore();
+          this.displayTasks(tasksList);
+        }
+        li.classList.remove('edit');
+      });
+
+      del.addEventListener('click', (e) => {
+        e.stopPropagation();
+        this.removeTask(index);
+        this.tasks.forEach((task, index) => {
+          task.index = index + 1;
+        });
+        this.setStore();
+        this.displayTasks(tasksList);
+      });
+
+      li.append(check, field, inputField, del, dots);
+      taskslist.appendChild(li);
+    });
+  }
+
+  addTask = (description) => {
+    if (description.trim()) {
+      const task = new Task(description.trim(), this.tasks);
+      this.tasks.push(task);
+    }
+  }
+
+  clearAll = () => {
+    this.tasks = [];
```

Comment:
Even though you are resetting the live array storing data, how are you updating local storage withing this function.

Reply:
The local storage has a separate function that updates it